symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/message/MessageService.java

```diff
@@ -206,18 +231,46 @@ public V4Message send(@Nonnull String streamId, @Nonnull String template, Object
    * @see <a href="https://developers.symphony.com/restapi/reference#create-message-v4">Create Message v4</a>
    */
   public V4Message send(@Nonnull String streamId, @Nonnull String message) {
+    return send(streamId, message, null, null);
+  }
+
+  /**
+   * Sends a message to the stream ID passed in parameter.
+   *
+   * @param streamId    the ID of the stream to send the message to
+   * @param message     the message payload in MessageML
+   * @param data        the data to be send with the message
+   * @param attachment  the attachment of the message
+   * @return a {@link V4Message} object containing the details of the sent message
+   * @see <a href="https://developers.symphony.com/restapi/reference#create-message-v4">Create Message v4</a>
+   */
+  public V4Message send(@Nonnull String streamId, @Nonnull String message, String data, File attachment) {
```

Comment:
I'd prefer to have an `InputStream attachment` or/and `byte[] attachment` instead of `File attachment`. 

`byte[]` is optional as developers can still easily `new ByteArrayInputStream(bytes)`

Reply:
In method implementation, the ephemeral `File` object would be temporary created in `java.io.tmpdir` folder. Could you however take a look to the `org.openapi.generator` plugin to see if there is way for API clients to accept something else than a File object.

Reply:
the problem here is that I don't think we can get the file type from an `InputStream` object. If we don't specify the extension file type in the file name, we will get the 403 Exception `File type not allowed`

Reply:
but how can we know by what we should replace`file-name.ext` since we get no information about the filename from the `InputStream`?

Reply:
Then developer should provide it, or at least just the extension (name could be just `UUID.randomUUID()`), I don't know

Reply:
Spring has the concept of MultipartFile that has a file name, a mime type and an input stream
Maybe we could have this sort of object to give us more flexibility (with constructors taking file, bytes,...)